paddle.chunk support unevenly division 易用性提升

NKNaN opened the pull request:
<!-- TemplateReference: https://github.com/PaddlePaddle/Paddle/wiki/PULL-REQUEST-TEMPLATE--REFERENCE -->
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->

### PR Category
<!-- One of [ User Experience | Execute Infrastructure | Operator Mechanism | CINN | Custom Device | Performance Optimization | Distributed Strategy | Parameter Server | Communication Library | Auto Parallel | Inference | Environment Adaptation ] -->
User Experience

### PR Types
<!-- One of [ New features | Bug fixes | Improvements | Performance | BC Breaking | Deprecations | Docs | Devs | Not User Facing | Security | Deprecations | Others ] -->
Improvements

### Description
<!-- Describe what you’ve done -->
https://pytorch.org/docs/2.5/generated/torch.chunk.html#torch.chunk
参照 torch.chunk 支持 x.shape[axis] 不能被 chunks 整除的情况。

NKNaN committed:
update

zhwesky2010 reviewed (commented):
这个是之前内部的讨论结论：

结论：对于非整除功能，有size优先（保证平均切分，牺牲个数）和个数优先（保证个数，牺牲平均度）两种设计方式，torch.split第二个参数字面上与split_size相关（size优先合理），torch.chunk第二个参数字面上与split_num相关（个数优先合理），但目前torch这两个API都是按size优先的思路实现。

而paddle.split、paddle.chunk为别名完全一样，第二个参数名字面上都与split_num相关，按**个数优先开发**更合理，但这样开发了也无法和torch完全对齐。

目前是按 哪种思路开发的？

NKNaN commented:
> 这个是之前内部的讨论结论：
> 
> 结论：对于非整除功能，有size优先（保证平均切分，牺牲个数）和个数优先（保证个数，牺牲平均度）两种设计方式，torch.split第二个参数字面上与split_size相关（size优先合理），torch.chunk第二个参数字面上与split_num相关（个数优先合理），但目前torch这两个API都是按size优先的思路实现。
> 
> 而paddle.split、paddle.chunk为别名完全一样，第二个参数名字面上都与split_num相关，按**个数优先开发**更合理，但这样开发了也无法和torch完全对齐。
> 
> 目前是按 哪种思路开发的？

目前是按照size优先改的，为了和 torch.chunk 对齐。如果要按个数优先开发，应该可以和 torch.tensor_split 对齐？

zhwesky2010 commented:
> 这个是之前内部的讨论结论：
> 
> 结论：对于非整除功能，有size优先（保证平均切分，牺牲个数）和个数优先（保证个数，牺牲平均度）两种设计方式，torch.split第二个参数字面上与split_size相关（size优先合理），torch.chunk第二个参数字面上与split_num相关（个数优先合理），但目前torch这两个API都是按size优先的思路实现。
> 
> 而paddle.split、paddle.chunk为别名完全一样，第二个参数名字面上都与split_num相关，按**个数优先开发**更合理，但这样开发了也无法和torch完全对齐。
> 
> 目前是按 哪种思路开发的？

@NKNaN 那这个就先不开发吧，感觉如果指定了chunks=num，就是指定了个数，但是最后不是按个数来的，设计上有点不合理。`torch.split`是合理的，`torch.chunk`这样设计不太合理。

NKNaN commented:
> > 这个是之前内部的讨论结论：
> > 结论：对于非整除功能，有size优先（保证平均切分，牺牲个数）和个数优先（保证个数，牺牲平均度）两种设计方式，torch.split第二个参数字面上与split_size相关（size优先合理），torch.chunk第二个参数字面上与split_num相关（个数优先合理），但目前torch这两个API都是按size优先的思路实现。
> > 而paddle.split、paddle.chunk为别名完全一样，第二个参数名字面上都与split_num相关，按**个数优先开发**更合理，但这样开发了也无法和torch完全对齐。
> > 目前是按 哪种思路开发的？
> 
> @NKNaN 那这个就先不开发吧，感觉如果指定了chunks=num，就是指定了个数，但是最后不是按个数来的，设计上有点不合理。`torch.split`是合理的，`torch.chunk`这样设计不太合理。

好的
